### PR TITLE
Handle new 'rustc -Z ls' output lines

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: src/compiler/rust.rs [hunk 1]
```diff
@@ -2895,13 +2897,36 @@ c:/foo/bar.rs:
 
     #[cfg(feature = "dist-client")]
     #[test]
-    fn test_parse_rustc_z_ls() {
+    fn test_parse_rustc_z_ls_old() {
```
Comment (mitchhentges) [inline src/compiler/rust.rs:2900]: I'd recommend using an absolute identifier to describe the difference in behaviour that you're testing here: "old" and "new" is relative, and will mean something different when these tests are investigated in, say, 2 years time.

Perhaps something like `test_parse_rustc_ls_pre_1_60()` (or, whichever rust version changed the behaviour).

**Review by mitchhentges — APPROVED** [review]
Makes sense, thanks for the patch.
I'm going to hold off on landing this until after the [`v0.3.0` release](https://github.com/mozilla/sccache/milestone/3)

File: src/compiler/rust.rs (hunk 1 above)
Comment (lissyx) [reply src/compiler/rust.rs:2900]: I'm not sure in which version it changed, from what I tested at least 1.58. Maybe I can find it within `rustc` source ...

Comment (lissyx) [reply src/compiler/rust.rs:2900]: https://github.com/rust-lang/rust/commit/cef3ab75b12155e0582dd8b7710b7b901215fdd6

**Comment by lissyx** [comment]
> Makes sense, thanks for the patch. I'm going to hold off on landing this until after the [`v0.3.0` release](https://github.com/mozilla/sccache/milestone/3)

Why wait ? This is already broken for m-c users with sccache-dist.

**Comment by mitchhentges** [comment]
Two reasons:
* I'm already undergoing testing/benchmarking for the current status of `main`
* There will be a follow-up release shortly after `v0.3.0` which will include improvements such as this one.

**Comment by lissyx** [comment]
Right but it's not an improvement, it's a fix required to be able to continue to use `sccache-dist` on current m-c.

**Comment by mitchhentges** [comment]
That's fair 👍

**Pull request merged**